qiskit/visualization/pulse_v2/generators.py

```diff
@@ -0,0 +1,683 @@
+
+import numpy as np
+
+from qiskit import pulse
+from qiskit.pulse import instructions
+from qiskit.visualization.exceptions import VisualizationError
+from qiskit.visualization.pulse_v2 import drawing_objects, data_types, PULSE_STYLE
+
+
+# Waveform related information generation
+
+
+def _parse_waveform(inst_data: data_types.InstructionTuple) \
+        -> Tuple[np.ndarray, np.ndarray, Dict[str, Any]]:
+    r"""A helper function that generates sample data array of the waveform with
+    instruction meta data.
+
+    Args:
+        inst_data: Instruction data set
+
+    Raises:
+        VisualizationError: When invalid instruction type is loaded.
+
+    Returns:
+        A tuple of xy data and metadata dictionary.
+    """
+    inst = inst_data.inst
+
+    meta = OrderedDict()
+    if isinstance(inst, instructions.Play):
+        # pulse
+        if isinstance(inst.pulse, pulse.ParametricPulse):
+            pulse_data = inst.pulse.get_sample_pulse()
+            meta.update(inst.pulse.parameters)
+        else:
+            pulse_data = inst.pulse
+        xdata = np.arange(pulse_data.duration) + inst_data.t0
+        ydata = pulse_data.samples
+    elif isinstance(inst, instructions.Delay):
+        # delay
+        xdata = np.arange(inst.duration) + inst_data.t0
+        ydata = np.zeros(inst.duration)
+    elif isinstance(inst, instructions.Acquire):
+        # acquire
+        xdata = np.arange(inst.duration) + inst_data.t0
+        ydata = np.ones(inst.duration)
+        acq_data = {'memory slot': inst.mem_slot.name,
+                    'register slot': inst.reg_slot.name if inst.reg_slot else 'N/A',
+                    'discriminator': inst.discriminator.name if inst.discriminator else 'N/A',
+                    'kernel': inst.kernel.name if inst.kernel else 'N/A'}
+        meta.update(acq_data)
+    else:
+        raise VisualizationError('Instruction %s cannot be drawn by filled envelope.' % type(inst))
+
+    meta.update({'duration (cycle time)': inst.duration,
+                 'duration (sec)': inst.duration * inst_data.dt if inst_data.dt else 'N/A',
+                 't0 (cycle time)': inst_data.t0,
+                 't0 (sec)': inst_data.t0 * inst_data.dt if inst_data.dt else 'N/A',
+                 'phase': inst_data.frame.phase,
+                 'frequency': inst_data.frame.freq,
+                 'name': inst.name})
+
+    return xdata, ydata, meta
+
+
+def _fill_waveform_color(channel: pulse.channels.Channel) \
+        -> data_types.ComplexColors:
+    r"""A helper function that returns color code of the fill waveform.
+
+    Args:
+        channel: Pulse channel object associated with the fill waveform.
+
+    Raises:
+        VisualizationError: When invalid channel is specified.
+
+    Returns:
+        A color code of real and imaginary part of the waveform.
+    """
+    if isinstance(channel, pulse.DriveChannel):
+        colors = PULSE_STYLE.style['formatter.color.fill_waveform_d']
```

Comment:
How does this work?

Reply:
`PULSE_STYLE` is initialized with the default stylesheet when we load the module. Then user can update `PULSE_STYLE` with stylesheet (this will be done in the interface so we don't have the actual code in this PR). `PULSE_STYLE.style` is just a flattened python dictionary.